vignettes/including-static-dashboards.Rmd

```diff
@@ -0,0 +1,126 @@
+
+```
+ui <- fluidPage(
+    uiOutput("dashboard")
+)
+```
+
+This creates the UI object, but most of the work is going to be done on the server side. We can use the `shinyUser()` function to get a user record for the current viewer of the app. This record includes some useful pieces of information, including the user's name, their email address, and their Crunch preferences. You can use this object to conditionally display different dashboards for different users like so: 
+
+```
+server <- function(input, output, session) {
+    user <- shinyUser()
+    output$dashboard <- renderUI({
+        if (is.character(email(user()))) {
+            includeHTML("acme_dashboard.html")
+        } else {
+            h1("You are not logged in, please visit app.crunch.io to login")
+        }
+    })
+}
+```
+
+This prevents unauthorized users from seeing the rendered dashboard, and directs them to the crunch website which will refresh their authentication cookie. 
+
+### Conditional dashboards and user identification
+
+We can take this pattern one step further to show different users different dashboards. We use the same UI object, but check the user's email and change the displayed HTML file based on that email. 
+
+```
+server <- function(input, output, session) {
+    user <- shinyUser()
+    output$dashboard <- renderUI({
+        if (grepl("acme.com$", email(user()))) {
+            includeHTML("acme_dashboard.html")
+        } else if (grepl("globex.com$", email(user()))) {
+            includeHTML("globex_dashboard.html")
+        } else {
+            h1("You are not logged in, please visit app.crunch.io to login")
+        }
+    })
+}
+```
+What this does is check the user's email address to identify clients from one company, and then displays a rendered dashboard for that company. If they are not logged into Crunch, they will get a message instructing them to log in. An example app of this kind is  [here](https://github.com/Crunch-io/crunchy/tree/master/inst/example_apps/rmd_dashboard_app), to run it, copy the app folder, open `app.R` and run it with `shiny::runApp("app.R")`, or by clicking the "Run App" button in RStudio. 
+
+### Managing Dashboard Renders
+
+The above code assumes that you are going to be re-rendering your Rmd files manually, and uploading the results to the Jupyter server. You can also re-render the Rmd files dynamically as part of the Shiny app by calling `rmarkdown::render()` somewhere in your server function. For example the following app re-renders the dashboard whenever the user clicks an action button. Note that for this to work smoothly you should use the [reactivePoll](https://shiny.rstudio.com/reference/shiny/1.0.2/reactivePoll.html) to ensure that the UI is refreshed after the file is regenerated. If you don't use `reactivePoll` you might end up in a situation where the UI refreshes before the RMarkdown file is finished rendering. 
+
+```
+library(crunchy)
+
+ui <- sidebarLayout(
+         sidebarPanel(
```

Comment:
This is also weirdly indented (at least as far as GitHub shows). I fixed the indentation in the `inst/` example.